src/main/java/org/zalando/nakadi/controller/ExceptionHandling.java

```diff
@@ -85,4 +87,18 @@ private String generateErrorTraceId() {
         }
         throw exception.getException();
     }
+
+    @ExceptionHandler(RepositoryProblemException.class)
+    public ResponseEntity<Problem> handleRepositoryProblem(final RepositoryProblemException exception,
+                                                           final NativeWebRequest request) {
+        LOG.error("Repository problem occurred", exception);
+        return Responses.create(Response.Status.SERVICE_UNAVAILABLE, exception.getMessage(), request);
+    }
+
+    @ExceptionHandler(InconsistentStateException.class)
```

Comment:
@v-stepanov if you want to have some specific exceptions processed in specific way you can have exception handler on the controller level. Another way is to define global exception handler for MyNakadiRuntimeException1. Wdyt?